yjit/src/codegen.rs

```diff
@@ -6567,6 +6568,9 @@ fn gen_send_cfunc(
                 perf_call!("gen_send_cfunc: ", known_cfunc_codegen(jit, asm, ci, cme, block, argc, recv_known_class))
             };
 
+            let method_id = unsafe { (*cme).called_id };
+            CompilationLog::add_cfunc(recv_known_class, method_id);
```

Comment:
I still don't understand the motivation to include the logs of C method calls, especially when you don't provide a convenient (instead of letting the user match "\<cfunc\>") to filter them out.

There's an issue related to this project in our team repository, which is titled "Tooling to detect dynamic code generation". We want to find Ruby code that is dynamically generated and compiled by YJIT. C method calls are just details of what's compiled in such Ruby code, so it doesn't seem to help you "detect" dynamic code generation.

Reply:
I guess this was my misunderstanding as well. I thought we'd want to include it in the log because it generates code at runtime. If that's not useful it's easy to remove.

Reply:
> I thought we'd want to include it in the log because it generates code at runtime.

You just need to log `gen_single_block` and drop this `gen_send_cfunc` logging. Every code generation goes through `gen_single_block`, and `gen_send_cfunc` is just one of the functions that `gen_single_block` may call. If the purpose is to cover everything that "generates code at runtime", you only need to log `gen_single_block`.

Reply:
There's more to what I had in mind, but it doesn't particularly matter. I've followed your guidance and dropped logging this event.